Ports TGMC bullet hit and blood splatter effects (gods least semi-modular PR)

maaacha opened the pull request:
## About The Pull Request

ports the blood splatter and bullet hit particle effects as seen in inserted video.

has presets for sparks, wood, rock, glass, leaf and snow, which have to be set on the add_debris_element proc on each object/turf. other presets can be made and added pretty easily.


https://github.com/LethalStation/NovaSector/assets/116771811/712ddb00-f130-43f9-b52b-f47bfe022c35



## How This Contributes To The Nova Sector Roleplay Experience

makes gunfights epicerer and cinematicerer

## Proof of Testing

<!-- Include any screenshots/videos/debugging steps of the code functioning successfully, between the </summary> and </details> code blocks. -->
<!-- To our mappers and spriters: Posting screenshots of content INSIDE EDITORS (aseprite, PDN, SDMM, ect) is NOT valid proof of testing. Please make sure that you COMPILE the game and provide PROOF you tested your edits. -->

<details>
<summary>Screenshots/Videos</summary>
  
</details>

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: Added new mechanics or gameplay changes
add: Added more things
del: Removed old things
qol: made something easier to use
balance: rebalanced something
fix: fixed a few things
sound: added/modified/removed audio or sound effects
image: added/modified/removed some icons or images
spellcheck: fixed a few typos
code: changed some code
refactor: refactored some code
config: changed some config setting
admin: messed with admin stuff
server: something server ops should know
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

maaacha committed:
testing shit

maaacha committed:
fikses

Ephemeralis commented:
Exquisite. We just need to grab their bullet sprite or one of our own and we're golden I think.

maaacha committed:
muzzle flash and smoke

maaacha commented:
added muzzle flash and smoke effects. muzzle flash and light are determined by values on the gun, i added color variants for the modular energy weapons and modified the flash values for the lethal guns but i havent touched any other guns (idk whats in rotation :3c)



https://github.com/LethalStation/NovaSector/assets/116771811/b10a549d-f69b-43b8-b056-60b63215e4bb

maaacha committed:
oopsie

maaacha committed:
unit test fix
